File: scripts/learn/__tests__/fixtures/clra-fixture/py/shapes.py
"""shapes.py — inheritance chain ground truth (mirrors src/shapes.ts).

    Circle(Shape)   → inherits edge

Python has no interfaces, so there is no `implements` mirror here; the
implements edge is TS-only (see src/shapes.ts).
"""


class Shape:
    def area(self):
        return 0


class Circle(Shape):
    def __init__(self, r):
        self.r = r

    def area(self):
        return 3.14 * self.r * self.r

File: scripts/learn/__tests__/fixtures/clra-fixture/src/shapes.ts
// shapes.ts — inheritance chain + interface implementation ground truth.
//   Circle extends Shape            → inherits edge
//   Circle implements Drawable      → implements edge
// Methods (area/draw) are intentionally OUT OF SCOPE for the dead-code oracle,
// which covers module-level functions only (see README.md § Oracle scope).

export interface Drawable {
  draw(): string;
}

export class Shape {
  area(): number {
    return 0;
  }
}

export class Circle extends Shape implements Drawable {
  constructor(private r: number) {
    super();
  }

  area(): number {
    return 3.14 * this.r * this.r;
  }

  draw(): string {
    return "circle";
  }
}
